src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -17,11 +17,14 @@ public abstract class FunctionInvokerBase : IFunctionInvoker, IDisposable
         private FileSystemWatcher _fileWatcher;
         private bool _disposed = false;
 
-        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata)
+        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata, ITraceWriterFactory traceWriterFactory = null)
         {
             Host = host;
             Metadata = functionMetadata;
-            TraceWriter = CreateTraceWriter(host.ScriptConfig, functionMetadata.Name);
+
+            traceWriterFactory = traceWriterFactory ?? new FunctionTraceWriterFactory(functionMetadata.Name, Host.ScriptConfig);
+            TraceWriter = traceWriterFactory.Create();
+            PrimaryHostTraceWriter = new ConditionalTraceWriter(TraceWriter, () => Host.IsPrimary);
```

Comment:
Is there a way for us to not have the two trace writers hanging off of the base class, and instead bake all this into the single TraceWriter we use everywhere?

I was envisioning that that single instance would be a ConditionalTraceWriter. In places where you need to log primary only logs, you would call an overload like `TraceWriter.Info("My message", primaryOnly: true)` which would add a property to the TraceEvent.Properties bag to filter on. Or perhaps `TraceWriter.Info(TraceEvent.Info("My message", new { primaryOnly: true }))`. You get the idea.